packages/taquito-michelson-encoder/src/tokens/bls12-381-fr.ts

```diff
@@ -0,0 +1,62 @@
+import { Token, TokenFactory, TokenValidationError } from './token';
+
+export class Bls12381frValidationError extends TokenValidationError {
+  name: string = 'Bls12381frValidationError';
+  constructor(public value: any, public token: Bls12381frToken, message: string) {
+    super(value, token, message);
+  }
+}
+export class Bls12381frToken extends Token {
+  // An element of the BLS12-381 scalar field Fr
+  // see https://tezos.gitlab.io/michelson-reference/#type-bls12_381_fr
+  static prim = 'bls12_381_fr';
+
+  constructor(
+    protected val: { prim: string; args: any[]; annots: any[] },
+    protected idx: number,
+    protected fac: TokenFactory
+  ) {
+    super(val, idx, fac);
+  }
+
+  private isValid(val: any): Bls12381frValidationError | null {
+    if (typeof val === 'string' && /^[0-9a-fA-F]*$/.test(val) && val.length % 2 === 0) {
+      return null;
+    } else {
+      return new Bls12381frValidationError(val, this, `Invalid bytes: ${val}`);
```

Comment:
Would it provide more value if we add condition (out of above 3) on which validation failed to the returned error?

Reply:
I'll make the change, thank you!